File: Solution/CPP/Shamiksha/employee/src/employee.cpp

# include "../include/employee.h"
#include<string>
#include <iostream>


using namespace std;


//implementation of Employee class methods

Employee::Employee(){

    id = 0;
    name = "";
    salary = 0.0;
}

Employee::Employee( int empId, string& empName, double empSalary) {
    id = empId;
    name = empName;
    salary = empSalary;
}


void Employee::displayInfo() {
    cout << "Employee ID: " << id << endl;
    cout << "Name: " << name << endl;
    cout << "Salary: $" << salary << endl;
}

Employee::~Employee() {
    // Destructor implementation (if needed)
    cout << "Employee object with ID " << id << " is being destroyed." << endl;
}

File: Solution/CSharp/Sarthak/EmployeeManagementSystem/Employee.cs
namespace HR
{
    public class Employee
    {
        private int id;
        private string name;
        private double salary;
        private Address address;

        public int Id
        {
            get { return id; }
            set { id = value; }
        }
        public string Name
        {
            get { return name; }
            set { name = value; }
        }
        public double Salary
        {
            get { return salary; }
            set { salary = value; }
        }
        public Address Address
        {
            get { return address; }
            set { address = value; }
        }


        public Employee(int id, string name, double salary,Address theAddress)
        {
            this.id = id;
            this.name = name;
            this.salary = salary;
            this.address = theAddress;
        }

        public void Display()
        {
            Console.WriteLine("ID: " + id + ", Name: " + name + ", Salary: " + salary);
            if (address != null)
            {
                address.Display();
            }
        }
        

    }
   
 }
